subworkflows/nf-core/fastq_removeadapters_merge/meta.yml

```diff
@@ -0,0 +1,144 @@
+keywords:
+  - adapters
+  - removal
+  - short reads
+  - merge
+  - trim
+components:
+  - trimmomatic
+  - cutadapt
+  - trimgalore
+  - bbmap/bbduk
+  - leehom
+  - fastp
+  - adapterremoval
+input:
+  - ch_reads:
+      type: file
+      description: |
+        List of FastQ files of size 1 and 2 for single-end and paired-end data, respectively.
+        Structure: [ val(meta), [ path(reads) ] ]
+  - skip_trimmomatic:
+      type: boolean
+      description: |
+        If true, skips the trimmomatic process.
+  - skip_cutadapt:
+      type: boolean
+      description: |
+        If true, skips the cutadapt process.
+  - skip_trimgalore:
+      type: boolean
+      description: |
+        If true, skips the trimgalore process.
+        We suggest running either cutadapt or trimgalore,
+        because trimgalore also runs cutadapt in the background.
+  - skip_bbduk:
+      type: boolean
+      description: |
+        If true, skips the bbmap/bbduk process.
+  - contaminants:
+      type: file
+      description: |
+        Reference files used by bbmap/bbduk and/or fastp, containing adapter and/or contaminant sequences for removal
+  - skip_leehom:
+      type: boolean
+      description: |
+        If true, skips the leehom process.
+  - skip_fastp:
+      type: boolean
+      description: |
+        If true, skips the fastp process.
+  - fastp_discard_trimmed_pass:
+      type: boolean
+      description: |
+        Specify true to not write any reads that pass trimming thresholds from the fastp process.
+        This can be used to use fastp for the output report only.
+  - fastp_save_trimmed_fail:
+      type: boolean
+      description: |
+        Specify true to save files that failed to pass fastp trimming thresholds
+        ending in `*.fail.fastq.gz`
+  - save_merged:
+      type: boolean
+      description: |
+        Specify true to save all merged reads to a file ending in `*.merged.fastq.gz`
+        Used by fastp
+  - skip_adapterremoval:
+      type: boolean
+      description: |
+        If true, skips the adapterremoval process.
+  - text_adapters:
+      type: file
+      description: Optional text file containing list of adapters to look for for removal
+        with one adapter per line. Otherwise will look for default adapters (see AdapterRemoval
+        man page), or can be modified to remove user-specified adapters via ext.args.
+output:
+  - reads:
+      type: file
+      description: The trimmed/modified single or paired end fastq reads
+      pattern: "*.fastq.gz"
+  - merged_reads:
+      type: file
+      description: |
+        The trimmed/modified fastq reads merged into a single file.
+        Only meaningful for paired-end reads.
+      pattern: "*.fastq.gz"
+  - discarded_reads:
+      type: file
+      description: |
+        All files containing discarded reads throughout the steps.
+      pattern: "*.fastq.gz"
+  - trimmomatic_trim_log:
```

Comment:
For some of these outputs where the underlying tool is optional, should these outputs be marked as optional?